src/logic/shorten-url.js

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import fetch from 'node-fetch';
+
+import { config } from '../config';
+import { getLogger } from '../log';
+
+class BitlyConfigurationError extends Error {
```

Comment:
Should we collect all the errors in `utils/errors.js`?

Reply:
I'm thinking of utils/errors to gather errors that are global to the program. Here it's clearly an error that's specific to this file.

I took this approach from the Result/Error approach in Rust, where each code unit implements errors specific to their business code.